### ISSUE TITLE
Master and stack layout issue when increasing master windows with only two windows open does not work.

### ISSUE THREAD

**Issue by sadikeey** [hyprwm/Hyprland#1836]
I'm not sure whether this behavior is a bug or a feature, but when using the master and stack layout, increasing the number of master windows does not take effect when only two windows are open. However, when more than two windows are open, it works perfectly fine.

Because in DWM and most other window managers, increasing the number of master windows while using the master and stack layout works as expected, regardless of the number of windows that are currently open.

Here are my keybindings that I am using to increase no. of masters windows

```sh
bind = $mainMod, i, layoutmsg, addmaster
bind = $mainMod, d, layoutmsg, removemaster
```

**Comment by vaxerski** [comment]
uhh what would you expect to happen? Two masters and an empty half of the screen?

**Comment by sadikeey** [comment]
What should happen is that if there are only two windows available, they should take up the full screen and be split in half horizontally not vertically.
 Like this

![Screenshot_2023-03-20_20:32:26](https://user-images.githubusercontent.com/86684667/226382251-09566882-1858-465f-8edc-12f3251f4464.png)

**Comment by vaxerski** [comment]
feel free to open a MR, I mostly do not touch master anymore apart from bugfixes because I do not use it and y'all have the weirdest workflows.

**Comment by Person1873** [comment]
Not seeing any action on this so far, I'll see if i can get this working.

**Comment by sadikeey** [comment]
> Not seeing any action on this so far, I'll see if i can get this working.

You can get ideas for implementing this feature using this hyprland plugin (https://github.com/hyprwm/Hyprland/issues/1836#issuecomment-1502954251)

**Comment by Person1873** [comment]
> feel free to open a MR, I mostly do not touch master anymore apart from bugfixes because I do not use it and y'all have the weirdest workflows.

I've opened a pull request for this, you're not wrong about us having weird workflows.
you should see multi workspace tagging

**Comment by Person1873** [comment]
Functionality has been merged, see https://github.com/hyprwm/Hyprland/pull/2025

**Comment by sadikeey** [comment]
> Functionality has been merged, see #2025

Thanks @Person1873
